api-analytics/service-worker.js

```diff
@@ -11,19 +11,30 @@ self.onactivate = function(event) {
   event.waitUntil(self.clients.claim());
 };
 
-// Fetch is as simply as log the request and passthrough.
-// Water clear thanks to the promise syntax!
 self.onfetch = function(event) {
-  event.respondWith(log(event.request).then(fetch));
+  event.respondWith(
+    // Log the request ...
+    log(event.request)
+    // .. and then actually perform it.
+    .then(fetch)
+  );
 };
 
 // Post basic information of the request to a backend for historical purposes.
 function log(request) {
-  var returnRequest = function() { return Promise.resolve(request); };
-  var data = { method: request.method, url: request.url };
+  var returnRequest = function() {
+    return request;
```

Comment:
Despite I know here it's not necessary, I prefer the form returning the Promise because of two main reasons:
  1. It saves the normalization step.
  2. It's more versatile, allowing it to appear at the beginning (`returnRequest().then(...);`) or in the middle of the chain (`.then(returnRequest).then(...);`)

Not blocking on this.